cpu/em_tuned.py

```diff
@@ -0,0 +1,213 @@
+                             'desktop', 'enterprise-storage',
+                             'laptop-ac-powersave',
+                             'laptop-battery-powersave', 'latency-performance',
+                             'network-latency', 'network-throughput',
+                             'powersave', 'server-powersave',
+                             'throughput-performance',
+                             'spindown-disk', 'virtual-guest', 'virtual-host']
+        available_profiles = self.get_available_profiles()
+        for profile in available_profiles:
+            if profile not in expected_profiles:
+                self.log.info("Profile: %s is not found" % profile)
+
+        active_profile = self.get_active_profile()
+        found = False
+        self.log.info(" Active profile : %s" % active_profile)
+        for profile in available_profiles:
+            if active_profile == profile:
+                found = True
+        if not found:
+            self.error_count += 1
+            self.log.info("Active profile: %s not found in the list of"
+                          "available profiles" % active_profile)
+
+    def profile_change(self):
+        """
+        Check all available profiles by change to active
+        """
+        available_profiles = self.get_available_profiles()
+        for profile in available_profiles:
+            self.set_active_profile(profile)
+            active_profile = self.get_active_profile()
+            if profile != active_profile:
+                self.error_count += 1
+                self.log.info("Profile: %s is not able to set" % profile)
+
+    def check_tuned_off(self):
+        """
+        Check for Tuned off
+        """
+        preserve_active_profile = self.get_active_profile()
+        self.set_tuned_off()
+        active_profile = self.get_active_profile()
+        if active_profile:
+            self.error_count += 1
+            self.log.info("Active profile: %s is present even after turning"
+                          "tuned off" % active_profile)
+        self.set_active_profile(preserve_active_profile)
+
+    # Helper functions
+    def set_tuned_off(self):
+        """
+        Turn off tuned
+        """
+        status = process.system("tuned-adm off", shell=True)
+        self.log.info(" status %s " % status)
+        if status != 0:
+            self.error_count += 1
+            self.log.info("Error turning off tuned %s" % status)
+
+    def set_active_profile(self, profile):
+        """
+        To set active profile
+        """
+        status = process.system("tuned-adm profile %s" % profile, shell=True)
+        if status != 0:
+            self.error_count += 1
+            self.log.info("Error setting Profile %s" % profile)
+
+    def get_available_profiles(self):
+        """
+        To get all available profiles
+        """
+        cmd_profile_list = "tuned-adm list|grep '^\-'|\
+                           sed 's/^- //' | awk '{print $1}'"
```

Comment:
Can we use python alternative for shell command? like
`
cmd_profile_list = []
cmd_profile = commands.getoutput('tuned-adm list').split('\n')
for line in cmd_profile:
        if line startswith("- "):
               cmd_profile.append[line]
`